compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java

```diff
@@ -10208,6 +10212,149 @@ private Boolean isFollowTypedBindingPattern(SyntaxKind tokenKind) {
+            }
+
+            bindingPatterns.add(mappingBindingPatternRhs);
+            mappingBindingPatternMember = parseMappingBindingPatternMember();
+            bindingPatterns.add(mappingBindingPatternMember);
+            token = peek();
+        }
+
+        STNode closeBrace = parseCloseBrace();
+
+        // seperating out the rest-binding-pattern
+        STNode restBindingPattern;
+        if (mappingBindingPatternMember.kind == SyntaxKind.REST_BINDING_PATTERN) {
+            restBindingPattern = bindingPatterns.remove(bindingPatterns.size() - 1);
+        } else {
+            restBindingPattern = STNodeFactory.createEmptyNode();
+        }
+
+        STNode bindingPatternsNode = STNodeFactory.createNodeList(bindingPatterns);
+        endContext();
+        return STNodeFactory.createMappingBindingPatternNode(openBrace, bindingPatternsNode, restBindingPattern,
+                closeBrace);
+    }
+
+    /**
+     * Parse mapping-binding-pattern entry.
+     *
+     * mapping-binding-pattern := { field-binding-patterns }
+     * field-binding-patterns :=
+     *    field-binding-pattern (, field-binding-pattern)* [, rest-binding-pattern]
+     *    | [ rest-binding-pattern ]
+     * field-binding-pattern :=
+     *    field-name : binding-pattern
+     *    | variable-name
+     *
+     * @return mapping-binding-pattern node
+     */
+    private STNode parseMappingBindingPatternMember() {
+        STToken token = peek();
+        switch (token.kind) {
+            case ELLIPSIS_TOKEN:
+                return parseRestBindingPattern();
+            default:
+                return parseFieldBindingPattern();
+        }
+    }
+
+    private STNode parseMappingBindingpatternRhs() {
+        return parseMappingBindingpatternRhs(peek().kind);
+    }
+
+    private STNode parseMappingBindingpatternRhs(SyntaxKind nextTokenKind) {
+        switch (nextTokenKind) {
+            case COMMA_TOKEN:
+                return parseComma();
+            case CLOSE_BRACE_TOKEN:
+                return null;
+            default:
+                Solution solution = recover(peek(), ParserRuleContext.MAPPING_BINDING_PATTERN_END_OR_CONTINUE);
+
+                // If the parser recovered by inserting a token, then try to re-parse the same
+                // rule with the inserted token. This is done to pick the correct branch
+                // to continue the parsing.
+                if (solution.action == Action.REMOVE) {
+                    return solution.recoveredNode;
+                }
+
+                return parseMappingBindingpatternRhs(solution.tokenKind);
+        }
+    }
+
+    /**
+     * Parse field-binding-pattern.
+     *
+     * field-binding-pattern := field-name : binding-pattern | varname
+     *
+     * @return field-binding-pattern node
+     */
+    private STNode parseFieldBindingPattern() {
+        startContext(ParserRuleContext.FIELD_BINDING_PATTERN);
+        STNode fieldName = parseVariableName();
```

Comment:
Since this can be either field-name or variable-name, we shouldn't use `parseVariableName()`.
Please refer to `parseWaitField()` and `parseReceiveField()`, those has a similar scenarios.